FaunaDB.Client/Types/RefV.cs

```diff
@@ -0,0 +1,159 @@
+﻿using System;
+using FaunaDB.Query;
+using Newtonsoft.Json;
+
+namespace FaunaDB.Types
+{
+    /// <summary>
+    /// A FaunaDB ref type.
+    /// <para>
+    /// See <see href="https://fauna.com/documentation/queries#values-special_types">FaunaDB Special Types</see>.
+    /// </para>
+    /// </summary>
+    public class RefV : Value
+    {
+        public string Id { get; }
+        public RefV Class { get; }
+        public RefV Database { get; }
+
+        public RefV(string id, RefV @class = null, RefV database = null)
+        {
+            Id = id;
+            Class = @class;
+            Database = database;
+
+            if (@class != null && !@class.IsClassRef())
+                throw new ArgumentException($"Argument '{nameof(@class)}' is not a class reference", nameof(@class));
+
+            if (database != null && !database.IsDatabaseRef())
+                throw new ArgumentException($"Argument '{nameof(database)}' it not a database reference", nameof(database));
+        }
+
+        protected internal override void WriteJson(JsonWriter writer)
+        {
+            var props = UnescapedObject.With(
+                "id", Id,
+                "class", Class,
+                "database", Database
+            );
+
+            writer.WriteObject("@ref", props);
+        }
+
+        public override bool Equals(Expr v)
+        {
+            var other = v as RefV;
+
+            return other != null &&
+                Id == other.Id &&
+                Class == other.Class &&
+                Database == other.Database;
+        }
+
+        protected override int HashCode() =>
+            new { Id, Class, Database }.GetHashCode();
+
+        public override string ToString()
+        {
+            var cls = Class != null ? $", class = {Class}" : string.Empty;
+            var db = Database != null ? $", database = {Database}" : string.Empty;
+
+            return $"{GetType().Name}(id = \"{Id}\"{cls}{db})";
+        }
+    }
+
+    static class RefExtensions
+    {
+        static bool IsBuiltIn(this RefV _ref) =>
+            _ref.Class == null && _ref.Database == null;
+
+        public static bool IsClassRef(this RefV cls)
```

Comment:
the method is public otherwise classes cannot access it, but the class is private